test/client_test.rb

```diff
@@ -175,21 +175,56 @@ class ClientTest < Minitest::Test
   end
 
   context "loading bins" do
-    should "succeed" do
-      stub_bins
+    setup do
+      # 1st page of 2 bins
+      stub_request(:get, %r{#{Vimaly::Client::VIMALY_ROOT_URL}/rest/2/company_id/bins\?max-results=2\&page-token=0}).to_return(
+        status: 200,
+        body: [{name: 'Alpha', _id: 1}, {name: 'Beta', _id: 2}].to_json
+      )
+      # 2nd page of 2 bins
+      stub_request(:get, %r{#{Vimaly::Client::VIMALY_ROOT_URL}/rest/2/company_id/bins\?max-results=2\&page-token=2}).to_return(
+        status: 200,
+        body: [{name: 'Gamma', _id: 3}, {name: 'Delta', _id: 4}].to_json
+      )
+      # 3rd page of 2 bins, returning only 1 (indicating the last page)
+      stub_request(:get, %r{#{Vimaly::Client::VIMALY_ROOT_URL}/rest/2/company_id/bins\?max-results=2\&page-token=4}).to_return(
+        status: 200,
+        body: [{name: 'Epsilon', _id: 5}].to_json
+      )
+    end
+
+    should "succeed and load all available bins" do
```

Comment:
Sweet, this test coverage LGTM, I think you're testing just about every scenario…barring a "failure scenario" wherein I think we just don't care if Vimaly goes boom?

Reply:
Well, I think if _vimaly_ goes 💥 this code here fails too, and we got notified right away. So no tests for that.